lib/cru_lib/global_registry_relationship_methods.rb

```diff
@@ -46,7 +46,11 @@ def create_in_global_registry(base_object, relationship_name)
 
       global_registry_id = Array.wrap(
         entity[base_object.class.global_registry_entity_type_name]["#{relationship_name}:relationship"]
-      ).detect { |hash| hash['client_integration_id'] == id.to_s }['relationship_entity_id']
+      ).detect do |hash|
+        cid = hash[:client_integration_id]
```

Comment:
Oops, not sure this has indifferent access. Safer to use hash['client_integration_id'] ?